android/src/main/kotlin/com/situm/flutter/ar/situm_ar/scene/ARSceneHandler.kt

```diff
@@ -88,84 +227,824 @@ class ARSceneHandler(
+        fenceModels[modelName] = situmARModel
+
+        val visiblePositions = getVisibleModelPositions(fenceModels)
+        val newPosition = generateValidPosition(
+            cameraPosition = sceneView.cameraNode.worldPosition,
+            cameraRotation = sceneView.cameraNode.worldRotation,
+            maxDistance = 5f,
+            heightOffset = height,
+            coneAngle = 30f,
+            existingPositions = visiblePositions,
+            minDistance = 2f
+        )
+        if (newPosition != null) {
+            modelNode.worldPosition = newPosition
+            modelNode.isVisible = true
+            sceneView.addChildNode(modelNode)
+        }
+    }
+
+
+    private fun parseGeofenceArMetadata(cf: Map.Entry<String, String>): List<Map<String, Any>> {
+        val json = JsonParser.parseString(cf.value.toString()).asJsonObject
+        val features = json["features"].asJsonArray
+
+        // Iterar por cada "Feature" y extraer la información requerida
+        val extractedData = features.map { feature ->
+            val properties = feature.asJsonObject["properties"].asJsonObject
+            val geometry = feature.asJsonObject["geometry"].asJsonObject
+
+            // Crear un mapa con los datos que queremos extraer
+            mapOf(
+                "name" to properties["name"].asString,
+                "url" to properties["url"].asString,
+                "scale" to properties["scale"].asFloat,
+                "coordinates" to geometry["coordinates"].asJsonArray.map { it.asFloat }
+            )
+        }
+        return extractedData
+    }
+
+
+    fun startModelProximityCheck() {
+        proximityCheckRunnable = object : Runnable {
+            override fun run() {
+                val userPosition = sceneView.cameraNode.worldPosition
+                val visiblePositions = getVisibleModelPositions(fenceModels)
+                val hasNearbyModels = visiblePositions.any { position ->
+                    distanceBetween(userPosition, position) < MIN_DISTANCE_TO_REGENERATE_MODELS
+                }
+
+                if (!hasNearbyModels) {
+                    regenerateModelsNearUser(userPosition)
+                }
+
+                handler.postDelayed(this, CHECK_MODELS_NEARBY_INTERVAL)
+            }
+        }
+        proximityCheckRunnable?.let { handler.post(it) }
+    }
+
+    fun stopModelProximityCheck() {
+        proximityCheckRunnable?.let { handler.removeCallbacks(it) }
+        proximityCheckRunnable = null
+    }
+
+    private fun regenerateModelsNearUser(userPosition: Position) {
+        fenceModels.forEach { (modelName, model) ->
+            if (model.modelNode.isVisible) {
+                updateExistingModel(model, model.modelNode.worldPosition.y)
+            }
+        }
+    }
+
+
+    fun isDebugMode(): Boolean {
+        return onDebug;
+    }
+
+    fun setDebugMode(debug: Boolean) {
+        onDebug = debug;
+    }
```

Comment:
Esta clase é un pouco peligro, pode ser? Ten xa máis de 1K liñas. Eu intentaría buscar a forma de estructuralo en clases máis pequenas con responsabilidades limitadas (xa sabes, principio de mínima responsabilidade). Non bloquearei a PR por esto, pero é o que eu faría.